src/FsCodec.SystemTextJson/UnionOrTypeSafeEnumConverterFactory.fs

```diff
@@ -5,12 +5,13 @@ open System.Text.Json.Serialization
 type UnionOrTypeSafeEnumConverterFactory(typeSafeEnum, union) =
     inherit JsonConverterFactory()
 
-    static let hasConverterAttribute: System.Type -> bool = memoize _.IsDefined(typeof<JsonConverterAttribute>, true)
+    static let cache = System.Collections.Concurrent.ConcurrentDictionary<System.Type, bool>()
+    static let typeHasConverterAttribute t: bool = cache.GetOrAdd(t, fun (t: System.Type) -> t.IsDefined(typeof<JsonConverterAttribute>, ``inherit`` = false))
```

Comment:
note flipping behavior to be less greedy (not honoring attributes on base types), to match Newtonsoft impl